jetty-websocket/websocket-common/src/main/java/org/eclipse/jetty/websocket/common/WebSocketSession.java

```diff
@@ -338,10 +337,9 @@ public void incomingFrame(Frame frame)
     public boolean isOpen()
     {
         if (this.connection == null)
-        {
             return false;
-        }
-        return !closed.get() && this.connection.isOpen();
+
+        return !onCloseCalled.get() && this.connection.isOpen();
```

Comment:
I do like that we now only have a single Atomic rather than two.

Reply:
The previous AtomicBoolean `closed` was never set to anything else so would always be false, so it looked like the intention was to use `onCloseCalled`.

If the Session has notified `onClose()` then it is not open but the connection may still be open.
Or the endpoint may have been closed but the `WebSocketConnection` has not yet been notified of the failure.

Reply:
I'd say that if the app has not yet been notified of the closer, then it's not an ISE exception to call suspend.    Yes there is a race there, but that's fundamental to the design of a suspend API, so it can't be fixed.

If onCloseCalled is false, but the connection is just about to close because of an async error, then that's just a race that can't be fixed.  Checking connection.isOpen() here makes the window a little smaller, but doesn't close it.     I'd say let's to try to fix a race with a race and the ISE for suspend should only be thrown if onCloseCalled is true - don't care if that is not in isOpen.

Reply:
Suspend now only checks `onCloseCalled` and does not use `isOpen()`.